### PR TITLE
Upgrade build system.

### LINKED ISSUE #1741: Options to case.build to speed incremental rebuilds
I feel like this was discussed earlier in one or more issues, but I can't find them, so I'm opening a new issue.

When I'm building new code that I've written, I usually have a lot of compilation errors initially (I know, it's probably just me...), and so I need to incrementally rebuild (rerunning ./case.build) a few times before the build is successful.

This currently takes quite a bit longer than it needs to, because before ./case.build gets to the library in question, it:

(a) Builds component namelists – sometimes twice, at least for some test types

(b) Checks whether all preceding libraries need to be rebuilt

These incremental rebuilds could be sped up if we had options to case.build to:

(1) skip namelist generation (addressing (a))

(2) build only a single, given library (addressing (b))... maybe this would imply also skipping namelist generation

What do people think about adding these or other options to speed up incremental rebuilds?

### PR DESCRIPTION
**Pull request by jgfouca**
1) Make incompatible flags to case.build mutually exclusive
2) Add new flag for building individual components
3) Never run buildnml when building individual libs/components

Test suite: by-hand, scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #1741 

User interface changes?: Yes, new -b flag to case.build.

Update gh-pages html (Y/N)?: N

Code review: jedwards, sacks

### CODE REVIEW

File: scripts/Tools/case.build [hunk 1]
```diff
@@ -49,18 +51,22 @@ OR
...
 
-    parser.add_argument("--clean", nargs="*", choices=allobjs,
-                        help="objects to clean"
-                        "if no arguments then clean all objects other than sharedlib objects")
+    mutex_group.add_argument("--clean", nargs="*", choices=allobjs,
+                             help="objects to clean"
+                             "if no arguments then clean all objects other than sharedlib objects")
+
+    mutex_group.add_argument("-b", "--build", nargs="+", choices=allobjs,
+                             help="libs to build")
```
Comment (billsacks) [inline scripts/Tools/case.build]: Please add documentation stating that the creation of namelists will be skipped if this argument is specified.

File: scripts/lib/CIME/build.py [hunk 2]
```diff
@@ -248,9 +254,8 @@ def _build_libraries(case, exeroot, sharedpath, caseroot, cimeroot, libroot, lid
                 if re.search("Current setting for", line):
                     logger.warn(line)
 
-
     # clm not a shared lib for ACME
-    if get_model() != "acme":
+    if get_model() != "acme" and not (buildlist is not None and "lnd" not in buildlist):
```
Comment (billsacks) [inline scripts/lib/CIME/build.py]: I'm finding the double (triple if you count "None") negatives here hard to parse. I'd find it easier to understand if this was refactored to the following (you should double-check this logic, though):

```python
if get_model() == "acme" or (buildlist is not None and "lnd" not in buildlist):
    pass
else
```

**Comment by jgfouca** [comment]
@billsacks Done.

**Comment by billsacks** [comment]
Thanks. @jgfouca I like your refactored condition better than my suggestion - thanks!

**Comment by jgfouca** [comment]
You're welcome. @jedwards4b this is ready to integrate.
